instrumentation/graphql/test/instrumentation/graphql/tracers/graphql_tracer_test.rb

```diff
@@ -0,0 +1,239 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../lib/opentelemetry/instrumentation/graphql'
+require_relative '../../../../lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer'
+
+describe OpenTelemetry::Instrumentation::GraphQL::Tracers::GraphQLTracer do
+  let(:graphql_tracer) { OpenTelemetry::Instrumentation::GraphQL::Tracers::GraphQLTracer }
+  let(:instrumentation) { OpenTelemetry::Instrumentation::GraphQL::Instrumentation.instance }
+  let(:exporter) { EXPORTER }
+  let(:spans) { exporter.finished_spans }
+  let(:config) { {} }
+
+  let(:query_string) do
+    <<-GRAPHQL
+      query($id: Int!){
+        simpleField
+        resolvedField(id: $id) {
+          originalValue
+          uppercasedValue
+        }
+      }
+    GRAPHQL
+  end
+
+  before do
+    exporter.reset
+    instrumentation.install(config)
+  end
+
+  after do
+    # Force re-install of instrumentation
+    instrumentation.instance_variable_set(:@installed, false)
+
+    # Reset various instance variables to clear state between tests
+    ::GraphQL::Schema.instance_variable_set(:@own_tracers, [])
+
+    # Reseting @graphql_definition is needed for tests running against version `1.9.x`
+    SomeOtherGraphQLAppSchema.remove_instance_variable(:@graphql_definition) if SomeOtherGraphQLAppSchema.instance_variable_defined?(:@graphql_definition)
+    SomeGraphQLAppSchema.remove_instance_variable(:@graphql_definition) if SomeGraphQLAppSchema.instance_variable_defined?(:@graphql_definition)
```

Comment:
There was some fairly tricky state leaking between tests, particularly for `1.9.x` version of GraphQL.  This is required to prevent duplicate tracers executing, as well as running the tests that would not otherwise have a tracer added.  This came up when adding the config options for `schemas`.